plugins/connection/docker.py

```diff
@@ -167,7 +181,7 @@ def _get_docker_version(self):
 
     def _get_docker_remote_user(self):
         """ Get the default user configured in the docker container """
-        p = subprocess.Popen([self.docker_cmd, 'inspect', '--format', '{{.Config.User}}', self._play_context.remote_addr],
+        p = subprocess.Popen([self.docker_cmd, 'inspect', '--format', '{{.Config.User}}', self.get_option('remote_addr')],
```

Comment:
@bcoca here `self.get_option('remote_addr')` always evaluates to `"inventory_hostname"`, which makes Ansible print `[WARNING]: unable to retrieve default user from docker container:   Error: No such object: inventory_hostname` for every task.

Reply:
the vars entry should ensure you don't get 'invenetory_hostname' but the actual value, but  that requires variables to be available, probably need a pc fallback

Reply:
This happened when it was called from `__init__()`, despite it said it should not be called from `__init__()` :) I removed that call.

I also made sure that extra args aren't repeatedly added, and that the user retrieval is cached (so it's not called multiple times per task).